check_pgactivity

```diff
@@ -552,24 +547,33 @@ sub find_hostkey($) {
 #  * the ref of the structure to save
 #  * the path to the file storage
 sub save($$$$) {
-    my $host    = shift;
-    my $name    = shift;
-    my $ref     = shift;
-    my $storage = shift;
-    my $all     = {};
-    my $hostkey = find_hostkey($host);
+    my $host     = shift;
+    my $name     = shift;
+    my $ref      = shift;
+    my $storage  = shift;
+    my $all      = {};
+    my $lockfile = "${storage}.lock";
+    my $hostkey  = find_hostkey($host);
 
+    open my $fh, '>', $lockfile or die "can't open «${lockfile}»: $!";
+    flock($fh, LOCK_EX) or die "can't get exclusive lock on «${lockfile}»: $!";
 
-    die "File «${storage}» not recognized as a check_pgactivity status file.\n\n"
-        ."Please, check its path or move away this wrong file"
-        if -r $storage and not is_storable $storage;
+    if ( -e $storage ) {
```

Comment:
I think we should add an extra `!-d` check to be sure that the given storage is not a directory and raise some explicit error in that case?  I don't see any reason to refuse the use of symlinks there so those should be fine.

Same for the related code in `load()`.

Reply:
I added an early check, among all others before calling the service itself.